src/components/SearchResults/AddPageToDatasetButton.js

```diff
@@ -0,0 +1,28 @@
+import { useResponsive } from 'hooks/useResponsive'
+import fromPairs from 'helpers/fromPairs'
+import { DatasetActionButton } from 'components/shared/DatasetActionButton'
+
+export const AddPageToDatasetButton = ({ dataToAdd }) => {
+  const { setResponsive } = useResponsive()
+
+  const downloadableSamples = fromPairs(
+    dataToAdd
+      .filter((item) => item.num_downloadable_samples > 0)
+      .map((item) => [
+        item.accession_code,
+        { all: true, total: item.num_downloadable_samples }
+      ])
+  )
```

Comment:
We should move the logic of creating this data structure to a method in `useDatasetAction.js`. Since it is only consumed there it will help understanding how those functions get that unique data structure.

This will mean an additional change in the hook to not to expect "preformatted" ie not expect `all` and `total` to be attributes on the passed in array.

I would expect something like this:
```suggestion
  const downloadableSamples = getFormattedDatasetData(
    dataToAdd
      .filter((item) => item.num_downloadable_samples > 0)
  )
```

Then in `useDatasetActions`
```js
const getFormattedDatasetData = (experimentList) =>
  Object.fromEntries(
    experimentList.map((item) => [
      item.accession_code,
      { all: true, total: item.num_downloadable_samples }
    ])
  )
```

Reply:
@davidsmejia, these implementations were approved in previous PRs https://github.com/AlexsLemonade/refinebio-web/pull/189 and https://github.com/AlexsLemonade/refinebio-web/pull/190 3 days ago and already merged into `dev`. 

Since this PR is for the move to dataset modal implementation, can we tackle this in a separate issue to keep the context as specific as possible? Please let me know, thank you David.

Reply:
I can create a new issue to update this tomorrow?

Reply:
Yeah can you apply my request to those as well in this PR after that this looks good to me.